zeebe/exporters/camunda-exporter/src/test/java/io/camunda/exporter/handlers/ListViewProcessInstanceFromProcessInstanceHandlerTest.java

```diff
@@ -307,15 +343,98 @@ void shouldUpdateEntityFromRecord() {
         .isEqualTo(processInstanceRecordValue.getParentElementInstanceKey());
     assertThat(processInstanceForListViewEntity.getJoinRelation())
         .isEqualTo(new ListViewJoinRelation(ListViewTemplate.PROCESS_INSTANCE_JOIN_RELATION));
-    assertThat(processInstanceForListViewEntity.getTreePath())
-        .isEqualTo("PI_" + processInstanceRecordValue.getProcessInstanceKey());
+    assertThat(processInstanceForListViewEntity.getTreePath()).isEqualTo("PI_111");
 
     // process name and version tag is read from the cache
     assertThat(processInstanceForListViewEntity.getProcessName()).isEqualTo("test-process-name");
     assertThat(processInstanceForListViewEntity.getProcessVersionTag())
         .isEqualTo("test-version-tag");
   }
 
+  @Test
+  public void shouldUpdateTreePathFromRecordWithCallActivity() {
+    // given
+    final long processDefinitionKey1 = 999L;
+    final Long pi1Key = 111L;
+    final Integer callActivityIndex1 = 3;
+    final String callActivityId1 = "callActivity1";
+    final long subprocess1Key = 987L;
+    final long callActivity1Key = 123L;
+    final long processDefinitionKey2 = 888L;
+    final Long pi2Key = 222L;
+    final Integer callActivityIndex2 = 1;
+    final String callActivityId2 = "callActivity2";
+    final long subprocess2Key = 876L;
+    final long callActivity2Key = 234L;
+    final Long pi3Key = 333L;
+    final Record<ProcessInstanceRecordValue> processInstanceRecord =
+        createProcessInstanceRecord(
+            List.of(
+                List.of(pi1Key, subprocess1Key, callActivity1Key),
+                List.of(pi2Key, subprocess2Key, callActivity2Key),
+                List.of(pi3Key)),
+            List.of(callActivityIndex1, callActivityIndex2),
+            List.of(processDefinitionKey1, processDefinitionKey2, 777L),
+            pi3Key);
+    processCache.put(
+        processDefinitionKey1,
+        new CachedProcessEntity(null, null, List.of("0", "1", "2", callActivityId1)));
+
+    processCache.put(
+        processDefinitionKey2, new CachedProcessEntity(null, null, List.of("0", callActivityId2)));
+
+    // when called process 3rd level
+    final ProcessInstanceForListViewEntity processInstanceForListViewEntity3 =
+        new ProcessInstanceForListViewEntity().setId(String.valueOf(pi3Key));
+    underTest.updateEntity(processInstanceRecord, processInstanceForListViewEntity3);
+    // then
+    assertThat(processInstanceForListViewEntity3.getTreePath())
+        .isEqualTo("PI_111/FN_callActivity1/FNI_123/PI_222/FN_callActivity2/FNI_234/PI_333");
```

Comment:
🔧 Maybe we call the variable entity to make it a bit more simpler.

❓ Understanding question on the treePath. Does this mean we will not contain a subprocess element instance key in the inter treePath?

Reply:
Exactly, we won't.